src/storage/heap_file.c

```diff
@@ -12237,19 +12260,111 @@ heap_attrinfo_transform_to_disk_internal (THREAD_ENTRY * thread_p, HEAP_CACHE_AT
       buf->ptr = PTR_ALIGN (buf->ptr, INT_ALIGNMENT);
     }
 
-  /* Record the length of the object */
-  new_recdes->set_record_length (ptr_varvals - buf->buffer);
+  if (ptr_varvals + mvcc_extra > buf->endptr)
+    {
+      return S_DOESNT_FIT;
+    }
+
+  *record_size = ptr_varvals - buf->buffer;
+
+  return S_SUCCESS;
+}
 
-  /* if not enough MVCC wasted space need to reallocate */
-  if (ptr_varvals + mvcc_wasted_space > buf->endptr)
+/*
+ * heap_attrinfo_transform_to_disk_internal () -
+ *                         Transform to disk an attribute information
+ *                         kind of instance.
+ *   return: SCAN_CODE
+ *           (Either of S_SUCCESS, S_DOESNT_FIT,
+ *                      S_ERROR)
+ *   attr_info(in/out): The attribute information structure
+ *   old_recdes(in): where the object's disk format is deposited
+ *   new_recdes(in):
+ *   lob_create_flag(in):
+ *
+ * Note: Transform the object represented by attr_info to disk format
+ */
+static SCAN_CODE
+heap_attrinfo_transform_to_disk_internal (THREAD_ENTRY * thread_p, HEAP_CACHE_ATTRINFO * attr_info,
+					  RECDES * old_recdes, record_descriptor * new_recdes, int lob_create_flag)
+{
+  OR_BUF buf;
+  bool is_mvcc_class;
+  size_t expected_size;
+  size_t record_size, header_size, offset_size;
+  size_t mvcc_extra;
+  SCAN_CODE status;
+  // *INDENT-OFF*
+  std::set<int> incremented_attrids;
+  // *INDENT-ON*
+
+  assert (new_recdes != NULL);
+
+  /* check to make sure the attr_info has been used, it should not be empty. */
+  if (attr_info->num_values == -1)
+    {
+      return S_ERROR;
+    }
+
+  /* get any of the values that have not been set/read */
+  if (heap_attrinfo_set_uninitialized (thread_p, &attr_info->inst_oid, old_recdes, attr_info) != NO_ERROR)
+    {
+      return S_ERROR;
+    }
+
+  /* start transforming the dbvalues into disk values for the object */
+  is_mvcc_class = !mvcc_is_mvcc_disabled_class (&(attr_info->class_oid));
+
+  /* determine the size */
+  expected_size = heap_attrinfo_determine_disksize (attr_info, is_mvcc_class, &offset_size);
+
+  mvcc_extra = 0;
+  header_size = OR_NON_MVCC_HEADER_SIZE;
+  if (is_mvcc_class)
+    {
+      header_size = OR_MVCC_INSERT_HEADER_SIZE;
+      mvcc_extra = OR_MVCC_DELETE_ID_SIZE + OR_MVCC_PREV_VERSION_LSA_SIZE;
+      if (old_recdes)
+	{
+	  header_size = OR_MVCC_INSERT_HEADER_SIZE + OR_MVCC_PREV_VERSION_LSA_SIZE;
+	  mvcc_extra = OR_MVCC_DELETE_ID_SIZE;
+	}
+      expected_size += OR_MVCC_MAX_HEADER_SIZE - OR_MVCC_INSERT_HEADER_SIZE;
+    }
+
+  record_size = 0;
+
+resize_and_start:
+  /* build record buffer */
+  new_recdes->resize_buffer (expected_size);
+  or_init (&buf, new_recdes->get_data_for_modify (), (int) expected_size);
+
+  /* build header */
+  status = heap_attrinfo_transform_header_to_disk (thread_p, attr_info, &buf, offset_size, is_mvcc_class, old_recdes);
```

Comment:
bool is_update = old_recdes != NULL; 변수 사용 검토; 이 변수로 대체할 수 있는 old_recdes 검사 부분에 일괄 적용 
```suggestion
  bool is_update = old_recdes != NULL;
  ...
  status = heap_attrinfo_transform_header_to_disk (thread_p, attr_info, &buf, offset_size, is_mvcc_class, is_update);
```

Reply:
반영했습니다. 깔끔하네요.